Fix rom/rtc.h deprecation compile warning for debug component

mmakaay opened the pull request:
# What does this implement/fix? 

The debug component triggers a compile warning:
```
In file included from src/esphome/components/debug/debug_component.cpp:8:
.../.platformio/packages/framework-espidf/components/esp32/include/rom/rtc.h:1:2:
warning: #warning rom/rtc.h is deprecated, please use esp32/rom/rtc.h instead
[-#warning rom/rtc.h is deprecated, please use esp32/rom/rtc.h instead
```

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [x] ESP32 IDF
- [x] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml
debug:
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

Commit:
Fix rom/rtc.h deprecation compile warning

Commit:
Restored old include for ESP32 Arduino.

davet2001 reviewed (commented):
The pattern used elsewhere in the file is 
````
#ifdef USE_ARDUINO
  //...
#elif defined(USE_ESP_IDF)
  //...
#endif
````
Maybe this should follow the same format?

Commit:
Use #elif instead of two #ifdef blocks.

davet2001 reviewed (approved):
LGTM 👍

mmakaay commented:
I don't directly see a way to do it unconditionally, unless it's by doing the conditional stuff in a utility header file only, and including that where needed. But that could lead to a lot of those utility header files.
With Arduino moving to IDF v4+ too, it'd be definitely better to base things on the IDF version that is used.

Commit:
Include correct .h file, based on IDF major version

mmakaay commented:
I modified the code to make use of the IDF major version to include the correct .h file.
Also removed the inclusion of `esp_idf_version.h` and moved up `esp_system.h`, because that include already includes the version header file.

A more generic solution seems over the top, because a quick grep on the code base shows me no other of these `rom/...` includes.

oxan commented:
I looked into getting the `esp32/rom/rtc.h` path to work unconditionally by adding the `include` directory onto the include path, but that turned out to be more complicated than just doing this.
